src/utils/safelyInvoke.spec.js

```diff
@@ -0,0 +1,20 @@
+import safelyInvoke from './safelyInvoke';
+
+describe('safelyInvoke', () => {
+  test('should return the original function', () => {
+    const myFunction = jest.fn();
+    const safeFunction = safelyInvoke(myFunction);
+    safeFunction();
+
+    expect(myFunction).toHaveBeenCalled();
+  });
+
+  test('should return an empty function if called with non-function parameter', () => {
+    const nonFunction = undefined;
+    expect(typeof safelyInvoke(nonFunction)).toBe('function');
+  });
+
+  test('should not throw errors when the provided empty function is called', () => {
+    expect(safelyInvoke(undefined)).not.toThrow();
```

Comment:
Actually, we have to call the function returned by safely invoke. And to test whether a function throws, we have to pass a callback function to jest and then invoke that function. If not it would be the same as 

```
expect(function(){}).not.toThrow();
```

As you can tell, it's not executing the function.

Reply:
It actually does execute the function. Try adding a `console.log` inside. Also see the Jest documentation for `.toThrow()`, some examples use functions by name without the function wrapper.

Reply:
Oh neat! I learned something new today! Thank you!